test/prompt/invocation_layer/test_hugging_face_inference.py

```diff
@@ -0,0 +1,266 @@
+    and that model_kwargs_rejected are correctly filtered out
+    """
+    model_kwargs = {"temperature": 0.7, "do_sample": True, "stream": True}
+    model_kwargs_rejected = {"fake_param": 0.7, "another_fake_param": 1}
+
+    layer = HFInferenceEndpointInvocationLayer(
+        model_name_or_path="google/flan-t5-xxl", api_key="some_fake_key", **model_kwargs, **model_kwargs_rejected
+    )
+    assert layer.model_input_kwargs == model_kwargs
+    assert len(model_kwargs_rejected) == 2
+
+
+@pytest.mark.unit
+def test_invoke_with_no_kwargs():
+    """
+    Test that invoke raises an error if no prompt is provided
+    """
+    layer = HFInferenceEndpointInvocationLayer(model_name_or_path="google/flan-t5-xxl", api_key="some_fake_key")
+    with pytest.raises(ValueError) as e:
+        layer.invoke()
+        assert e.match("No prompt provided.")
+
+
+@pytest.mark.unit
+def test_invoke_with_stop_words():
+    """
+    Test stop words are correctly passed to HTTP POST request
+    """
+    stop_words = ["but", "not", "bye"]
+    layer = HFInferenceEndpointInvocationLayer(model_name_or_path="google/flan-t5-xxl", api_key="fake_key")
+    with unittest.mock.patch(
+        "haystack.nodes.prompt.invocation_layer.HFInferenceEndpointInvocationLayer._post"
+    ) as mock_post:
+        # Mock the response, need to return a list of dicts
+        mock_post.return_value = MagicMock(text='[{"generated_text": "Hello"}]')
+
+        layer.invoke(prompt="Tell me hello", stop_words=stop_words)
+
+        assert mock_post.called
+
+        # Check if stop_words are passed to _post as stop parameter
+        called_args, called_kwargs = mock_post.call_args
+        assert "stop" in called_kwargs["data"]["parameters"]
+        assert called_kwargs["data"]["parameters"]["stop"] == stop_words
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("stream", [True, False])
+def test_streaming_stream_param_in_constructor(stream):
+    """
+    Test stream parameter is correctly passed to HTTP POST request via constructor
+    """
+    layer = HFInferenceEndpointInvocationLayer(
+        model_name_or_path="google/flan-t5-xxl", api_key="fake_key", stream=stream
+    )
+    with unittest.mock.patch(
+        "haystack.nodes.prompt.invocation_layer.HFInferenceEndpointInvocationLayer._post"
+    ) as mock_post:
+        # Mock the response, need to return a list of dicts
+        mock_post.return_value = MagicMock(text='[{"generated_text": "Hello"}]')
+        layer.invoke(prompt="Tell me hello")
+
+    assert mock_post.called
+
+    # Check if stop_words are passed to _post as stop parameter
+    called_args, called_kwargs = mock_post.call_args
+
+    # stream is always passed to _post
+    assert "stream" in called_kwargs
+
+    # Check if stream is True, then stream is passed as True to _post
+    if stream:
+        assert called_kwargs["stream"]
+    # Check if stream is False, then stream is passed as False to _post
+    else:
+        assert not called_kwargs["stream"]
```

Comment:
This way is much better, we avoid the `if` and the code is similar to `test_streaming_stream_param_in_method`.

```suggestion
    assert called_kwargs["stream"] == stream
```